src/server.c

```diff
@@ -2812,6 +2815,11 @@ int isContainerCommandBySds(sds s) {
     return has_subcommands;
 }
 
+struct redisCommand *lookupSubcommand(struct redisCommand *container, sds sub_name) {
```

Comment:
the subcommand dict should have the fullanme as keys, not the sub_name
(there's used to be `lookSubcommandByFullanme` - where is it?)

Reply:
also this function can be a one-liner:
```
return dictFetchValue(container->subcommands_dict, sub_name);
```
but maybe it shouldn't even be a function? when we bring back `lookSubcommandByFullanme` the only place we need to search by sub_name is in `lookupCommandLogic` and then we don't even need a function...

Reply:
i thought in here, we agree to use the `declared_name` as the subcommand dict key?
https://github.com/redis/redis/pull/10127#discussion_r788086208

Reply:
yes, I'm sorry, I misunderstood.

i've deleted/updated some of my comments

Reply:
please check my last five commit.. hope there will be no other problems this time :)
btw, thanks for your patience and the all explanation and the reviews